src/common/types.cpp

```diff
@@ -503,6 +509,7 @@ bool LogicalType::IsIntegral() const {
 	case LogicalTypeId::UINTEGER:
 	case LogicalTypeId::UBIGINT:
 	case LogicalTypeId::HUGEINT:
+	case LogicalTypeId::UUID:
```

Comment:
While UUID is stored as a number, it isn't a "normal" integer number so probably doesn't belong in this list, e.g. normal arithmetic and such is not supported for UUIDs.